### PR TITLE
Clear Dynamic claims cache when role or organization are changed.

### PR DESCRIPTION
**Pull request by maliming**
If a role changes its name or is removed or an organization adds users or roles, we need to clear the dynamic claims cache for the users belonging to a role or an organization.

### CODE REVIEW

File: modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/OrganizationUnitEntityUpdatedOrDeletedEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        IOrganizationUnitRepository organizationUnitRepository,
+        IDistributedCache<AbpDynamicClaimCacheItem> cache)
+    {
+        Logger = NullLogger<OrganizationUnitEntityUpdatedOrDeletedEventHandler>.Instance;
+
+        _organizationUnitRepository = organizationUnitRepository;
+        _cache = cache;
+    }
+
+    public virtual async Task HandleEventAsync(EntityUpdatedEventData<OrganizationUnit> eventData)
```
Comment (rqx110) [inline modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/OrganizationUnitEntityUpdatedOrDeletedEventHandler.cs]: forget [UnitOfWork]?

File: modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/OrganizationUnitManager.cs [hunk 2]
```diff
@@ -189,9 +198,17 @@ await OrganizationUnitRepository.GetAsync(ouId, true)
...
-    public virtual Task RemoveRoleFromOrganizationUnitAsync(IdentityRole role, OrganizationUnit organizationUnit)
+    public virtual async Task RemoveRoleFromOrganizationUnitAsync(IdentityRole role, OrganizationUnit organizationUnit)
     {
         organizationUnit.RemoveRole(role.Id);
-        return OrganizationUnitRepository.UpdateAsync(organizationUnit);
+        await OrganizationUnitRepository.UpdateAsync(organizationUnit);
+        await RemoveDynamicClaimAsync(organizationUnit);
+    }
+
+    protected virtual async Task RemoveDynamicClaimAsync(OrganizationUnit organizationUnit)
```
Comment (rqx110) [inline modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/OrganizationUnitManager.cs]: i suggest to rename with `RemoveDynamicClaimsFromCacheAsync`

File: modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/UserEntityUpdatedOrDeletedEventHandler.cs [hunk 3]
```diff
@@ -35,6 +41,7 @@ public virtual async Task HandleEventAsync(EntityDeletedEventData<IdentityUser>
 
     protected virtual async Task ClearAsync(Guid userId, Guid? tenantId)
```
Comment (rqx110) [inline modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/UserEntityUpdatedOrDeletedEventHandler.cs]: i suggest to rename with `RemoveDynamicClaimsFromCacheAsync`

**Comment by maliming** [comment]
Move to https://github.com/abpframework/abp/pull/18200

**Pull request closed without merging**